sklearn/metrics/regression.py

```diff
@@ -272,7 +276,15 @@ def median_absolute_error(y_true, y_pred):
                                                    'uniform_average')
     if y_type == 'continuous-multioutput':
         raise ValueError("Multioutput not supported in median_absolute_error")
-    return np.median(np.abs(y_pred - y_true))
+    if sample_weight is None:
+        return np.median(np.abs(y_pred - y_true))
+    else:
+        check_consistent_length(y_pred, sample_weight)
+        sample_weight = np.array(sample_weight)
+        y_pred = y_pred.ravel()
+        y_true = y_true.ravel()
+        return weighted_median(np.abs(y_pred - y_true),
+                                    np.asarray(sample_weight))
```

Comment:
Oops I converted it twice.. I did it because it can be a python array..will change it..thanks..